build-tools/automation/azure-pipelines.yaml

```diff
@@ -231,3 +232,204 @@ stages:
         artifactName: win-build-test-results
         targetPath: $(Build.ArtifactStagingDirectory)
       condition: always()
+
+- stage: test
+  displayName: Test
+  dependsOn: mac_build
+  jobs:
+  - job: mac_apk_tests
+    displayName: APK Instrumentation
+    pool: $(XA.Build.Mac.Pool)
+    timeoutInMinutes: 240
+    cancelTimeoutInMinutes: 5
+    variables:
+      ApkTestConfiguration: Release
+    steps:
+    - task: DownloadPipelineArtifact@1
+      inputs:
+        artifactName: $(InstallerArtifactName)
+        itemPattern: "*.pkg"
+        downloadPath: $(System.DefaultWorkingDirectory)
+
+    - template: yaml-templates/run-installer.yaml
+
+    - task: MSBuild@1
+      displayName: build xaprepare
+      inputs:
+        solution: build-tools/xaprepare/xaprepare.sln
+        configuration: $(XA.Build.Configuration)
+        msbuildArguments: /t:Restore,Build
+
+    - script: |
+        mono build-tools/xaprepare/xaprepare/bin/$(XA.Build.Configuration)/xaprepare.exe --s=UpdateMono --auto-provision=yes --auto-provision-uses-sudo=yes --no-emoji --run-mode=CI
+        mono build-tools/xaprepare/xaprepare/bin/$(XA.Build.Configuration)/xaprepare.exe --s=Required --auto-provision=yes --auto-provision-uses-sudo=yes --no-emoji --run-mode=CI
+        mono build-tools/xaprepare/xaprepare/bin/$(XA.Build.Configuration)/xaprepare.exe --s=AndroidToolchain --no-emoji --run-mode=CI
+      displayName: provision dependencies
+
+    - task: NuGetCommand@2
```

Comment:
This task and the subsequent `build Xamarin.Android.Tools.BootstrapTasks.csproj` feel "brittle" to me, in that we're "duplicating" the (starting to get deeply) nested dependency tree between `.csproj` files and `azure-pipelines.yaml`.

Would it work to instead *add* `Xamarin.Android.Tools.sln` and `Xamarin.Android.Tools.BootstrapTasks.csproj` to `Xamarin.Android-Tests.sln` so that when we restore and built it (below), we also restore these two implicit dependencies it has?  (Thus turning implicit dependencies into more explicit ones?)

Reply:
I don't think we want to add the various submodule external source dependencies to the test solution, as it complicates things when working locally. I do believe we should look to clean this up a bit however in the future, we should be able to simplify our test dependencies.